discretize/TreeMesh.py

```diff
@@ -98,23 +99,48 @@ class TreeMesh(BaseTensorMesh, InnerProducts, TreeMeshIO):
 
     _meshType = 'TREE'
 
-    def __init__(self, h, x0=None, levels=None):
+    levels = properties.Integer(
+        "discretization level",
+        min=0
+    )
+
+    cells = properties.Set(
```

Comment:
I would prefer to keep this a private variable `_cells`. See discussion on aranzgeo/properties#191